tools/gen_fresh_dirinfo.py

```diff
@@ -195,6 +209,73 @@ def is_address_port_changed(
     return False
 
 
+def terminate_process(process: Popen) -> None:
+    """
+    Terminate a process in a clean way.
+
+    :param process: process to terminate
+    """
+    process.terminate()
+    try:
+        process.communicate(timeout=3)
+    except TimeoutExpired:
+        process.kill()
+
+
+def call_tor_gencert(
+        password: bytes,
+        create_new_identity: bool,
+        reuse_signing_key: bool,
+        identity_key_file: Path,
+        signing_key_file: str,
+        certificate_file: str,
+        lifetime_in_months: int,
+    ) -> None:
+    """
+    Call the tor-gencert command to generate key pairs and certificate.
+
+    :param password: the password to encrypt the identity key
+    :param create_new_identity: create a new identity key
+    :param reuse_signing_key: reuse an existing signing key
+    :param identity_key_file: file where to read or write the identity key
+    :param signing_key_file: file where to read or write the signing key
+    :param certificate_file: file where to write the certificate
+    :param lifetime_in_months: lifetime of the certificate in months
```

Comment:
See - here the `lifetime_in_months` directly indicates the unit, which is very useful!

Reply:
Good point, I am going to rename the options with less ambiguous names.